cmd/kubectl-gadget/trace/tcpdump.go

```diff
@@ -0,0 +1,244 @@
+					decoderArgs = append(decoderArgs, strings.Split(decoderArgsParam, " ")...)
+				}
+				if decoderBinaryParam != "" {
+					decoderBinary = decoderBinaryParam
+				}
+
+				decoderCmd = exec.Command(decoderBinary, decoderArgs...)
+				decoderCmd.Stdout = os.Stdout
+				decoderCmd.Stderr = os.Stderr
+				decoderCmd.Stdin = r
+				err = decoderCmd.Start()
+				if err != nil {
+					return fmt.Errorf("could not start tcpdump: %w", err)
+				}
+
+				out = w
+			}
+
+			if out != nil {
+				var err error
+				dummyInterface := pcapgo.DefaultNgInterface
+				dummyInterface.LinkType = layers.LinkTypeEthernet
+				dummyInterface.SnapLength = uint32(snapLen)
+				ngw, err = pcapgo.NewNgWriterInterface(out, dummyInterface, pcapgo.NgWriterOptions{SectionInfo: pcapgo.NgSectionInfo{
+					Hardware:    runtime.GOARCH,
+					OS:          runtime.GOOS,
+					Application: "InspektorGadget",
+					Comment:     "using gopacket",
+				}})
+				if err != nil {
+					return fmt.Errorf("instantiating NgWriter: %w", err)
+				}
+				ngw.Flush()
+			}
+
+			filter := strings.Join(args, " ")
+
+			tcpdumpGadget := &TraceGadget[types.Event]{
+				name:        "tcpdump",
+				commonFlags: commonFlags,
+				parser:      NewTCPDump(&commonFlags.OutputConfig, filter, snapLen, decoder, ngw),
+				params: map[string]string{
+					types.FilterStringParam: filter,
+					types.SnapLenParam:      strconv.Itoa(snapLen),
+				},
+			}
+
+			return tcpdumpGadget.Run()
+		},
+	}
+
+	utils.AddCommonFlags(cmd, commonFlags)
+	cmd.Flags().StringVar(&decoderParam, "decoder", "internal", "name of the decoder to use (either tcpdump, wireshark, internal or file)")
+	cmd.Flags().StringVar(&decoderArgsParam, "decoder-args", "", "arguments to forward to decoder")
+	cmd.Flags().StringVar(&decoderBinaryParam, "decoder-binary", "", "path to decoder binary (defaults to 'wireshark' or 'tcpdump' depending on decoder)")
+	cmd.Flags().StringVar(&filenameParam, "out-file", "", "output file name")
+	cmd.Flags().IntVar(&snapLen, "snaplen", 68, "number of bytes to capture")
+	return cmd
+}
+
+func NewTCPDump(outputConfig *commonutils.OutputConfig, filter string, snapLen int, decoder Decoder, pcapngWriter *pcapgo.NgWriter) commontrace.TraceParser[types.Event] {
+	columnsWidth := map[string]int{}
+	outputConfig.OutputMode = commonutils.OutputModeCustom
+	return &TCPDumpParser{
+		BaseParser:   commonutils.NewBaseWidthParser[types.Event](columnsWidth, outputConfig),
+		filter:       filter,
+		snapLen:      snapLen,
+		decoder:      decoder,
+		pcapngWriter: pcapngWriter,
+		interfaces:   make(map[string]int),
+	}
+}
+
+func (p *TCPDumpParser) getPodInterface(event *types.Event) int {
+	p.interfacesLock.RLock()
+	if id, ok := p.interfaces[event.Container]; ok {
+		p.interfacesLock.RUnlock()
+		return id
+	}
+	p.interfacesLock.RUnlock()
+	// Define new interface
+	p.interfacesLock.Lock()
+	id, err := p.pcapngWriter.AddInterface(pcapgo.NgInterface{
+		Name:        event.Container,
```

Comment:
offtopic: Is there anyway to print this interface name in tcpdump? I found https://serverfault.com/questions/224698/how-to-display-interface-in-tcpdump-output-flow but it seems rather complex.

Reply:
tcpdump seems to not handle the additional information (well), I haven't found a clean solution either. WireShark does however fully support it.